### PR TITLE
more fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: src/crwimage_int.cpp [hunk 1]
```diff
@@ -1091,7 +1091,7 @@ namespace Exiv2 {
         case 0x0012: ifdId = canonPiId; break;
         }
         assert(ifdId != ifdIdNotSet);
-        DataBuf buf = packIfdId(image.exifData(), ifdId, pHead->byteOrder());
+        DataBuf&& buf = packIfdId(image.exifData(), ifdId, pHead->byteOrder());
```
Comment (piponazo) [inline src/crwimage_int.cpp:1094]: `packIfdId` is returning a normal object. Why are you using `&&` here?

Is it part of the automatic transformation or something that you added manually?

File: src/makernote_int.cpp [hunk 2]
```diff
@@ -483,7 +483,7 @@ namespace Exiv2 {
         // Todo: This removes any gap between the header and
         // makernote IFD. The gap should be copied too.
         TiffHeader th(byteOrder);
-        DataBuf buf = th.write();
+        const DataBuf& buf = th.write();
```
Comment (piponazo) [inline src/makernote_int.cpp:486]: Same question here.  The signature of write is:

```
        virtual DataBuf write() const;
```

Why to add the `const & ` ??

File: src/actions.cpp [hunk 3]
```diff
@@ -487,34 +487,25 @@ std::unique_ptr<Task> Task::clone() const {
     int Print::printMetadata(const Exiv2::Image* image)
     {
         bool ret = false;
-        bool noExif = false;
+        bool noExif = (Params::instance().printTags_ & Exiv2::mdExif) && image->exifData().empty();
```
Comment (piponazo) [inline src/actions.cpp:490]: 👍

File: src/crwimage_int.cpp (hunk 1 above)
Comment (neheb) [reply src/crwimage_int.cpp:1094]: manual. So the explicit change did more than the changes here. I reverted some of them. This was part of getting explicit working. I'll revert when I can.

File: src/makernote_int.cpp (hunk 2 above)
Comment (neheb) [reply src/makernote_int.cpp:486]: Same as above. I'll revert.

**Review by piponazo — APPROVED** [review]
Thanks! ❤️

**Pull request merged**